sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1217,6 +1217,32 @@ void ProgramManager::addImages(pi_device_binaries DeviceBinary) {
           Entry->second.initialize(TypeSize, DeviceImageScopeDecorated);
         }
       }
+      // ... and initialize associated host_pipe information
+      {
+        std::lock_guard<std::mutex> HostPipesGuard(m_HostPipesMutex);
+
+        auto HostPipes = Img->getHostPipes();
+        for (const pi_device_binary_property &HostPipe : HostPipes) {
+          auto Entry = m_HostPipes.find(HostPipe->Name);
+          assert(Entry != m_HostPipes.end() &&
+                 "Host pipe has not been registered.");
+
+          pi::ByteArray HostPipeInfo =
+              pi::DeviceBinaryProperty(HostPipe).asByteArray();
+
+          // The supplied host_pipe info property is expected to contain:
+          // * 8 bytes - Size of the property.
+          // * 4 bytes - Size of the underlying type in the host_pipe.
+          // * 1 byte  - 0 if host_pipe has device_image_scope and any value
+          //             otherwise.
+          // Note: Property may be padded.
+          assert(HostPipeInfo.size() >= 13 && "Unexpected property size");
+          const std::uint32_t TypeSize =
```

Comment:
Do not repeat the type
```suggestion
          auto TypeSize =
```